astropy/coordinates/spectral_coordinate.py

```diff
@@ -54,24 +55,20 @@ def _apply_relativistic_doppler_shift(scoord, velocity):
     # since we can't guarantee that their metadata would be correct/consistent.
     squantity = scoord.view(SpectralQuantity)
 
-    beta = velocity / c
-    doppler_factor = np.sqrt((1 + beta) / (1 - beta))
-
-    if squantity.unit.is_equivalent(u.m):  # wavelength
-        return squantity * doppler_factor
-    elif (
-        squantity.unit.is_equivalent(u.Hz)
-        or squantity.unit.is_equivalent(u.eV)
-        or squantity.unit.is_equivalent(1 / u.m)
-    ):
-        return squantity / doppler_factor
-    elif squantity.unit.is_equivalent(KMS):  # velocity
-        return (squantity.to(u.Hz) / doppler_factor).to(squantity.unit)
-    else:  # pragma: no cover
-        raise RuntimeError(
-            f"Unexpected units in velocity shift: {squantity.unit}. This should not"
-            " happen, so please report this in the astropy issue tracker!"
-        )
+    beta = float(velocity / c)
```

Comment:
Does the main speed up come from this? It may be as it avoids quantity calculations when one can make it dimensionless. This works, but only if velocity is a scalar, but ideally we would support arrays (which we may not test...). Do you still get a decent speed up if you do `beta = (velocity / c).to_value(u.one)`? That avoids the unit conversions but would still allow broadcasting.